Task#5 Authorization

alexandrtovmach opened the pull request:
Stack:
* passport
* passport-local

Created:
Login function
Logout function
isLogged middleware

alexandrtovmach committed:
cache commit

alexandrtovmach committed:
Merge branch 'dev' into test

alexandrtovmach committed:
auth done

lexa0303 commented:
Я же сделал сервис для юзеров, возможно стоит его использовать а не схему напрямую ?

И возможно сделать один AuthService в котором будут login и logout, а не два сервиса ? т.к. эти методы если и будут нужны то вместе.

И middleware наверное стоит подключать через app.use, а не app.all

alexandrtovmach committed:
corrected according to comments lexa0303

- remove dependency with UserSchema
- add dependency with userRepository
- add "findById" method to userRepository
- change "app.all" to "app.use" in server.js

alexandrtovmach commented:
fixed all exept second article

alexandrtovmach committed:
fix bug of checking password

alexandrtovmach committed:
Merge remote-tracking branch 'origin/mfsn-2' into msfn-5

alexandrtovmach committed:
fixed method of userSchema and his call from passportStrategy

alexandrtovmach commented:
finall, please check

alexandrtovmach committed:
decryptService done

alexandrtovmach committed:
Merge remote-tracking branch 'origin/msfn-26' into msfn-5

alexandrtovmach committed:
add decrypt to loginService

alexandrtovmach commented:
add synch with decryptService

alexandrtovmach committed:
fix bug

alexandrtovmach committed:
on decrypt

DmitriyDN committed:
Merge branch 'dev' into msfn-5

DmitriyDN committed:
Merge branch 'dev' into msfn-5
